pkg/api/v1alpha1/cluster_types.go

```diff
@@ -63,9 +63,28 @@ type ClusterSpec struct {
 	PodIAMConfig                *PodIAMConfig                `json:"podIamConfig,omitempty"`
 	Packages                    *PackageConfiguration        `json:"packages,omitempty"`
 	// BundlesRef contains a reference to the Bundles containing the desired dependencies for the cluster
-	BundlesRef *BundlesRef `json:"bundlesRef,omitempty"`
+	// Deprecated: Use EksaVersion instead.
+	BundlesRef  *BundlesRef  `json:"bundlesRef,omitempty"`
+	EksaVersion *EksaVersion `json:"eksaVersion,omitempty"`
 }
 
+// EksaVersion is the CLI version used to create/upgrade the cluster.
+type EksaVersion string
+
+// Equal checks if two EksaVersions are equal.
+func (n *EksaVersion) Equal(o *EksaVersion) bool {
+	if n == o {
+		return true
+	}
+	if n == nil || o == nil {
+		return false
+	}
+	return *n == *o
+}
+
+// DevEksaVersion can be used in tests.
+var DevEksaVersion EksaVersion = "v0.0.0-dev"
```

Comment:
Can we move this somewhere in the `internal/test` folder instead then?

Reply:
yes